src/makeCalculator.js

```diff
@@ -4,7 +4,39 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const object = {
+    result: 0,
+  };
+
+  object.add = function (number) {
+    object.result += number;
+  };
+
+  object.subtract = function (number) {
+    object.result -= number;
+  };
+
+  object.multiply = function (number) {
+    object.result *= number;
+  };
+
+  object.divide = function (number) {
+    object.result /= number;
+  };
+
+  object.reset = function () {
+    object.result = 0;
+
+    return this;
+  };
+
+  object.operate = function (callback, number) {
+    callback(number);
+
+    return this;
+  };
+
+  return object;
```

Comment:
The variable name 'object' is too generic. Consider using a more descriptive name that reflects its role as a calculator, such as 'calculator'.